### DISCUSSION TITLE
Impossible to delete or uninstall BlackHole.driver

### DISCUSSION THREAD

**Discussion by ProfStringer** [ExistentialAudio/BlackHole#405] [Q&A]
Mac OS 11.3.1 (Big Sur) keeps warning that "BlackHole.driver" will damage your computer. You should move it to the Trash."  I've tried everything I know.  

- Obviously, clicking the "Move to Trash" doesn't work.  
- I've tried your uninstaller.  The fact that BlackHole is still running in the Sound > Input list and the Audio MIDI Setup panel with no way to quit or remove from either location would tell me that "Uninstallation was successful" is a lie
-  I've tried manually uninstalling via the Terminal per your exact instructions.  
- I've tried re-installing the latest version and repeating everything again.  

No matter what I do, BlackHole keeps reinstalling itself on every reboot and I cannot get rid of this fucking thing.  This is acting like a persistent rootkit.  No wonder Apple has designated it malware.

Since I'm trusting this isn't actually malware, but I no longer need it, what magic incarnation do I have to perform to get this thing off my system once and for all??????

[Here's a screenshot showing the steps taken and the fact that it won't quit.](https://i.imgur.com/eqc5t4J.jpg)

**Comment by devinroth** [comment]
Haha. It doesn’t reinstall. Probably just not getting uninstalled.

Send me screenshots of your HAL folder.

Devin Roth

**Reply by ProfStringer** [reply]
[Here's a screenshot showing the steps taken and the fact that it won't quit.](https://i.imgur.com/eqc5t4J.jpg)

**Comment by devinroth** [comment]
You it needs to be “BlackHole16ch.driver”

Try that first.


Devin Roth

**Reply by ProfStringer** [reply]
I don't understand.  It is the 16 channel version.  So were the rm commands I entered.

**Comment by devinroth** [comment]
Yeah you just missed the ch at the end. The rm command needs to match the file/folder you want to remove.

Devin Roth

**Reply by ProfStringer** [reply]
Slight progress. Thanks. But we're not there yet.

First, there's a typo on your Uninstall instructions. Your own command is missing the 'ch' on 'BlackHole16.driver'
https://github.com/ExistentialAudio/BlackHole/wiki/Uninstallation

After re-running with the missing 'ch', the processes did end in the Audio Devices and MIDI setup. However, on reboot, I still get the malware message, and I can see that BlackHole 16ch is still running under Multi-Output Device.

**Comment by devinroth** [comment]
Photos please.

Devin Roth

**Comment by daxmacrog** [comment]
I'm starting to think this is some sort of prank. As stated by the OP, it's impossible to delete BlackHole, and the steps listed in various posts don't work and reference files that don't exist. One uninstaller fails during its own install. The one for 2ch installs but then does nothing. I can't even find what it installed. But there's BlackHole, sitting in Audio MIDI Setup (with the minus button grayed out).

**Comment by devinroth** [comment]
The #1 cause of this is Rogue Amoeba’s software. They do some strange things with CoreAudio. If you have any of their software installed, remove it, restart your computer.

If BlackHoleXch.driver is not listed in the HAL folder there is no way that the computer can load BlackHole after restarting unless there is other software “Hijack”ing CoreAudio.

Devin

**Comment by daxmacrog** [comment]
Rogue Amoeba Audio Hijack was definitely the culprit! Strange. Thanks Devin!
